tests/import_proxy_test.go

```diff
@@ -0,0 +1,360 @@
+	ocpClient, err := configclient.NewForConfig(f.RestConfig)
+	Expect(err).ToNot(HaveOccurred())
+	proxy, err := ocpClient.ConfigV1().Proxies().Get(context.TODO(), controller.ClusterWideProxyName, metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		Skip("This OpenShift cluster version does not have a Cluster Wide Proxy object")
+	}
+
+	By("Updating OpenShift Cluster Wide Proxy with ImportProxy urls")
+	proxy.Spec.HTTPProxy = proxyHTTPURL
+	proxy.Spec.HTTPSProxy = proxyHTTPSURL
+	// disable proxing the OpenShift API calls
+	proxy.Spec.NoProxy = "*"
+	_, err = ocpClient.ConfigV1().Proxies().Update(context.TODO(), proxy, metav1.UpdateOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Wating OpenShift Cluster Wide Proxy reconcile")
+	// the default OpenShift no_proxy configuration only appears in the proxy object after and http(s) url is updated
+	Eventually(func() bool {
+		cwproxy, err := ocpClient.ConfigV1().Proxies().Get(context.TODO(), controller.ClusterWideProxyName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		if cwproxy.Status.HTTPProxy == proxyHTTPURL && cwproxy.Status.HTTPSProxy == proxyHTTPSURL {
+			return true
+		}
+		return false
+	}, time.Second*60, time.Second).Should(BeTrue())
+	proxy, err = ocpClient.ConfigV1().Proxies().Get(context.TODO(), controller.ClusterWideProxyName, metav1.GetOptions{})
+	Expect(err).ToNot(HaveOccurred())
+
+	By("Wating CDIConfig reconcile")
+	Eventually(func() bool {
+		config, err := f.CdiClient.CdiV1beta1().CDIConfigs().Get(context.TODO(), common.ConfigName, metav1.GetOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		http, _ := controller.GetImportProxyConfig(config, common.ImportProxyHTTP)
+		https, _ := controller.GetImportProxyConfig(config, common.ImportProxyHTTPS)
+		if http == proxy.Status.HTTPProxy && https == proxy.Status.HTTPSProxy {
+			// enable proxing the importer pod http or https calls
+			err := utils.UpdateCDIConfig(f.CrClient, func(config *cdiv1.CDIConfigSpec) {
+				config.ImportProxy = &cdiv1.ImportProxy{
+					HTTPProxy:  &http,
+					HTTPSProxy: &https,
+					NoProxy:    &noProxy,
+				}
+			})
+			Expect(err).ToNot(HaveOccurred())
+			return true
+		}
+		return false
+	}, time.Second*30, time.Second).Should(BeTrue())
+}
+
+// verifyImporterPodInfoInProxyLogs verifiy if the importer pod request (method, url and impoter pod IP) appears in the proxy log
+func verifyImporterPodInfoInProxyLogs(f *framework.Framework, dataVolume *cdiv1.DataVolume, isHTTPS bool, uri string, expected func() types.GomegaMatcher) {
+	Eventually(func() bool {
+		importerPod := getImporterPod(f)
+		proxyLog := getProxyLog(f)
+		return wasPodProxied(isHTTPS, uri, importerPod, proxyLog)
+	}, time.Second*60, time.Second).Should(expected())
+}
+
+func getImporterPod(f *framework.Framework) *corev1.Pod {
+	fmt.Fprintf(GinkgoWriter, "INFO: Waiting importer pod to become ready\n")
+	err := waitForPodIP(f, common.ImporterPodName, f.Namespace.Name, utils.PodWaitForTime)
```

Comment:
It seems we have failures in this particular part of the test in most lanes. I am trying to understand exactly what we are doing here and this is what I think we are doing:
1. Get the IP of the importer pod
2. Get the logs from the proxy pod (we can look up the proxy pod name at the start of the test)
3. Check if IP and URI.host and method are in a line in the proxy logs.

A few things I think we can improve:
1. You can specify a flag like --since or --since-time to kubectl logs to only return recent logs so you are not scanning a huge log.
2. We are only interested in the importer pod IP, once we have it, we don't have to look up the importer pod anymore.
3. The wait.PollImmediate is an old construct, we should be using Eventually instead.

Example of how I would look up the importer pod IP:
```go
podIp := ""
Eventually(func() string {
  pod, err := utils.FindPodByPrefix(f.K8sClient, namespace, podName, common.CDILabelSelector)
  if err != nil {
    return ""
  }
  fmt.Fprintf(GinkgoWriter, "INFO: Checking POD %s phase: %s\n", podName, string(pod.Status.Phase))
  podIp = pod.Status.PodIP != "" {
  return podIp
}, time.Second*60, time.Second).Should(NotBe(Empty()))
// podIp should be something at this point.
```
Now you have the IP address and you can look up the proxy logs.

Reply:
Yes, that is the goal of this method, and thanks for the feedback!

I found the problem of the test been failing intermittently.... Sometimes, the importer pod finishes too fast and the method cannot inspect the pod. To solve that, I decrease the import pod log pooling interval and changed the image size from the tiny image to the large one...

Also, I simplified some parts of the code to be cleaner.

Reply:
So we have that problem in other places as well and we have a special port on the http server that is rate limited, this ensures the import takes longer and thus the pod won't disappear as fast. Look for HTTPRateLimitPort for the port number, and TinyCoreQcow2URLRateLimit for an example of how to use it.

Reply:
Thanks, I updated it accordingly.

Reply:
Thanks, I updated it accordingly.